core/server/data/schema/schema.js

```diff
@@ -452,6 +452,8 @@ module.exports = {
         stats: {type: 'text', maxlength: 65535, nullable: true},
         email_count: {type: 'integer', nullable: false, unsigned: true, defaultTo: 0},
         subject: {type: 'string', maxlength: 300, nullable: true},
+        from: {type: 'string', maxlength: 191, nullable: true},
```

Comment:
The `maxlength` of 191 for `from` and `reply_to` make sense. Wanted to point out we don't seem to have any server side validation for the settings values, so this might be a source of potential problem is somebody goes crazy with their emails in settings.

Reply:
> don't seem to have any server side validation for the settings values

In terms of length or format? Format shouldn't matter here as this is only a record of what was attempted, if the _length_ can be anything then, yeah, that could be a problem 🤔 Especially as the "from" could be quite expand quickly... `"My really long site name"<crazyemail@reallylongdomain.com>`

Reply:
Both format and length. The length would only be limited by settings' value `maxlength: 65535`. 

Might be worth to expand this to something bigger, the site title is limited to [150](https://github.com/TryGhost/Ghost/blob/master/core/server/data/schema/default-settings.json#L61) plus 191 of potentially correct email address. 2000 might be too much for this situation, but we rarely use any values between 191 and 200 (og_title and og_description are 300 and 500 respectively). :shrug:

Reply:
We've tended to stick to these values for lengths:

```
 * Small strings = length 50
 * Medium strings = length 191
 * Large strings = length 1000-2000
 * Text = length 65535 (64 KiB)
 * Long text = length 1,000,000,000
```

> og_title and og_description are 300 and 500 respectively

I'm not entirely sure what the reasoning for those were. We have validations in place for 300/500 on the meta_* fields which are the fallback for these fields but those are soft limits rather than hard limits with the underlying columns having a length of 2000. Seems like the og/twitter_* columns are the exceptions.

If we're following convention then 2000 seems like the appropriate choice here for now knowing that exceptionally long site addresses or email addresses are likely invalid in any case and a symptom of not having per-setting validation for those values.

Reply:
Another exception 😬:  `posts_meta.email_subject` and `emails.subject` are both 300. That could also have problems in the future because the post's title is used as a fallback and that has a maxlength of 2000 although it's currently soft-limited via validation to 255 chars. Not a huge amount of consistency all round 🤷🏻‍♂️

I'll stick with 2000 for now as it's a "safe" value.

Reply:
Is it worth to add  `posts_meta.email_subject` and `emails.subject` to 4.0 migration list and redefine these columns with length 2000? :thinking:

Reply:
I think what we need first is a decision made for the absolute max length of an email subject, right now we have 255 and 300 as max lengths depending on which piece of data is used.

I'm not sure we should be taking the emails table maxlengths as a definition for the original data source because the emails table is just a record of values defined elsewhere. As long as the email table maxlengths are the same or longer than the max lengths of the source fields we should be fine.